tracing-attributes/src/lib.rs

```diff
@@ -216,42 +277,63 @@ pub fn instrument(args: TokenStream, item: TokenStream) -> TokenStream {
             Err(err) => return quote!(#err),
         };
 
-        let param_names: Vec<Ident> = params
+        let param_names: Vec<(Ident, Ident)> = params
             .clone()
             .into_iter()
             .flat_map(|param| match param {
                 FnArg::Typed(PatType { pat, .. }) => param_names(*pat),
                 FnArg::Receiver(_) => Box::new(iter::once(Ident::new("self", param.span()))),
             })
+            // if we are inside a function generated by async-trait, we should take care to rewrite
+            // "_self" as "self" for 'user convenience'
```

Comment:
I think the presence of a `_self` identifier is another likely marker that `async-trait` is in use...should that be part of our heuristic for detecting `async-trait` as well?

Reply:
You're right that we could do that to assert with more certainty that we are facing a aync_trait-like situation (aka: it is more probable that async_trait is in use if you see the the both pattern `fn f(...) Box::pin(f(...))` and `_self`) , but this is not necessary true will all traits, so its absence cannot rule out the use of async_trait. Consider this for example:
```rust
#[async_trait]
pub trait Foo {
    // the parameter v is here to see how asyc_trait rewrites variable names in the `impl` block
    async fn foo(v: usize) -> ();
}
```
This trait doesn't take `self`, so async_trait won't give us a `_self` parameter.

In the end, if we want to use this to be more precise, maybe we can consider that functions with the aforementioned pattern but with `self` as an argument are not cases of async_trait, as `self` should have rewritten it to `self`. But I wonder if there are other crates that exists and the does similar things without renaming `self` to `_self` or with a different name, and that we don't want to exclude simply because they don't apply the exact same mecanism as async_trait.

Reply:
That's fair, I think the pattern we're detecting is much more reliable.